contrib/windows-cmake/CMakeLists.txt

```
@@ -68,7 +86,16 @@ add_library(libhwloc
 
 set_target_properties(libhwloc PROPERTIES DEFINE_SYMBOL _USRDLL)
 
-target_include_directories(libhwloc PRIVATE ${TOPDIR}/include ${TOPDIR}/hwloc ${CMAKE_CURRENT_BINARY_DIR}/include)
+target_include_directories(libhwloc
+    PRIVATE ${TOPDIR}/include ${TOPDIR}/hwloc ${CMAKE_CURRENT_BINARY_DIR}/include
+    INTERFACE
+    "$<BUILD_INTERFACE:${TOPDIR}/include;${CMAKE_CURRENT_BINARY_DIR}/include>"
+    $<INSTALL_INTERFACE:include>
+)
+
+# imported target for use via FetchContent
+add_library(hwloc::hwloc INTERFACE IMPORTED)
+target_link_libraries(hwloc::hwloc INTERFACE libhwloc)
```

```suggestion
add_library(hwloc::libhwloc ALIAS libhwloc)
```

hwloc::hwloc is probably reserved for an executable target.

---

The hwloc::hwloc naming is standard for a library. I also fixed the naming (previously, it was making liblibhwloc.a (duplicated liblib).

---

alias targets are for special purposes, the imported target is more general object used for internal and external projects

---

> alias targets are for special purposes, the imported target is more general object used for internal and external projects

Alias targets are what they are named after or to quote cmake: "An ALIAS target is a name which may be used interchangeably with a binary target name in read-only contexts".
Imported targets however are meant to be imported external/existing targets or to quote the cmake docs "An IMPORTED target represents a pre-existing dependency.". 

You can use an alias target here without any issue and your intended usage here is just like the example mentioned by CMake in its docs about ALIAS targets:
https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#alias-targets

In the end it doesn't matter if it is an `IMPORTED` or an `ALIAS` target since both will achieve the same. I just think that `ALIAS` would be more correct in the usage here, would express intend more directly and also have only one line of code. 

> The hwloc::hwloc naming is standard for a library

unless there is an executable called hwloc which breaks the standard ;). I have seen other libraries do that.

---

changed to hwloc